apps/prairielearn/src/tests/instructorAssessmentSettings.test.ts

```diff
@@ -0,0 +1,325 @@
+  });
+
+  after(helperServer.after);
+
+  step('access the test assessment info file', async () => {
+    const assessmentInfo = JSON.parse(await fs.readFile(assessmentLiveInfoPath, 'utf8'));
+    assert.equal(assessmentInfo.title, 'Homework for file editor test');
+  });
+
+  step('change assessment info', async () => {
+    const settingsPageResponse = await fetchCheerio(
+      `${siteUrl}/pl/course_instance/1/instructor/assessment/1/settings`,
+    );
+    assert.equal(settingsPageResponse.status, 200);
+
+    const response = await fetchCheerio(
+      `${siteUrl}/pl/course_instance/1/instructor/assessment/1/settings`,
+      {
+        method: 'POST',
+        form: {
+          __action: 'update_assessment',
+          __csrf_token: settingsPageResponse.$('input[name="__csrf_token"]').val(),
+          orig_hash: settingsPageResponse.$('input[name="orig_hash"]').val(),
+          title: 'Test Title',
+          type: 'Homework',
+          set: 'Practice Quiz',
+          number: '1',
+          module: 'Module2',
+          aid: 'HW2',
+        },
+      },
+    );
+
+    assert.equal(response.status, 200);
+    assert.equal(response.url, `${siteUrl}/pl/course_instance/1/instructor/assessment/1/settings`);
+  });
+
+  step('verify assessment info change', async () => {
+    assessmentLiveInfoPath = path.join(assessmentLiveDir, 'HW2', 'infoAssessment.json');
+    const assessmentLiveInfo = JSON.parse(await fs.readFile(assessmentLiveInfoPath, 'utf8'));
+    assert.equal(assessmentLiveInfo.title, 'Test Title');
+    assert.equal(assessmentLiveInfo.type, 'Homework');
+    assert.equal(assessmentLiveInfo.set, 'Practice Quiz');
+    assert.equal(assessmentLiveInfo.number, '1');
+    assert.equal(assessmentLiveInfo.module, 'Module2');
+  });
+
+  step('verify nesting an assessment id', async () => {
+    const settingsPageResponse = await fetchCheerio(
+      `${siteUrl}/pl/course_instance/1/instructor/assessment/1/settings`,
+    );
+    assert.equal(settingsPageResponse.status, 200);
+
+    const response = await fetchCheerio(
+      `${siteUrl}/pl/course_instance/1/instructor/assessment/1/settings`,
+      {
+        method: 'POST',
+        form: {
+          __action: 'update_assessment',
+          __csrf_token: settingsPageResponse.$('input[name="__csrf_token"]').val(),
+          orig_hash: settingsPageResponse.$('input[name="orig_hash"]').val(),
+          title: 'Test Title',
+          type: 'Homework',
+          set: 'Practice Quiz',
+          number: '1',
+          module: 'Module2',
+          aid: 'nestedPath/HW2',
+        },
+      },
+    );
+
+    assert.equal(response.status, 200);
+    assert.equal(response.url, `${siteUrl}/pl/course_instance/1/instructor/assessment/1/settings`);
+  });
+
+  step('verify changing aid did not leave empty directories', async () => {
+    const assessmentDir = path.join(assessmentLiveDir, 'HW2');
+    assert.equal(await fs.pathExists(assessmentDir), false);
+  });
```

Comment:
These tests should be able to replace the tests removed from `courseEditor.test`.